src/signframework/cca_functions.c

```diff
@@ -803,8 +819,14 @@ int Digital_Signature_Verify(unsigned long signature_field_length,	/* input */
 
     exit_data_length = 0;			/* must be 0 */
 
-    rule_array_count = 1;
-    memcpy(rule_array,"PKCS-1.1", 8);		/* PKCS#1 padding */
+    if (SIGN_PKCS_PSS == signmode) {
+        rule_array_count = 2;
+        memcpy(rule_array,"PKCS-PSS", 8);		/* PKCS#1 with RSASSA-PSS scheme */
+        memcpy(rule_array+8,"SHA-512 ", 8);
```

Comment:
You are hard coding sha-512 with your SIGN_PKCS_PSS.
It might be worth changing your enum to reflect that behavior.

Reply:
I decided to just check the length is correct that way we could support different hash sizes in the future